### DISCUSSION TITLE
Merged Cell duplicate values instead of NONE

### DISCUSSION THREAD

**Discussion by barum** [jsvine/pdfplumber#623] [Ask for help with specific PDFs]
For merged cells (row or column), would it be possible to get the value into every cell that was merged? is this doable? or maybe an idea for a feature?

File:

[merged-cells.pdf](https://github.com/jsvine/pdfplumber/files/8235782/merged-cells.pdf)

```
import pdfplumber

def keep_visible_lines(obj):
    if obj['object_type'] == 'rect':
        return obj['non_stroking_color'] == 0
    return True

pdf = pdfplumber.open('./merged-cells.pdf')

page = pdf.pages[0]
page.filter(keep_visible_lines)

table_settings = {
    "horizontal_strategy": "lines",
    "vertical_strategy": "lines"
}
table = page.extract_table(table_settings)
print(table)


```
But the result is: 

```
[['Row-M1', 'Row-M2', 'column-merged'], ['R1-M', None, 'C1'], ['R2A', 'R2B', 'C2'], ['R3A', 'R3B', 'C3-M'], ['R4A', 'R4B', None], ['R-C5-M', None, 'C5-M']]
```

**Comment by jsvine** [comment]
Thanks for the suggestion, @barum! There's no super-easy way for users to do that now (though you can access the necessary coordinates/etc. via `page.find_tables(...)[0].cells`), but I've been thinking about how to improve this in the future. For reference, how would you *want* to see the result represented? Could you provide a reformatted version of the output that you'd like to see?

**Comment by barum** [comment]
Well. Because you know the Cell(s) that are merged, then all you have to do is take the cells and use the 1 value for all the merged cells.
example if you a value (YYYYY) in Column D and its merged in Row 1 & 2, then the data would be YYYYY for D1 and D2.
example if you have value(ZZZZZ) in Column A and B for ROW 3, the the data for A3 and B3 would be ZZZZZ

An option should be "treat_merged_Cell_value" = "merged-NONE", "merged-Same". If someone selects "none" then put NONE in all cells but 1. If someone selects "SAME", copy the data to all cells in that are merged.

Also, for any CELL that does NOT have value replace with "BLANK". this way there is a clear difference between normal blank cells and cells that were merged. 

does that help?

**Reply by jsvine** [reply]
Thanks, @barum. That's a great start, but I worry that there's still some nuance/ambiguity possible with this description (especially with respect to horizontally- vs. vertically-merged cells, and double-height cells). Could you provide the literal result you'd expect to see from this table extraction — i.e., what output you'd expect  from `print(table)` in your example above?

**Comment by barum** [comment]
given the Spreadsheet attached. I would expect to see the result as:

![Screen Shot 2022-03-16 at 12 27 11 PM](https://user-images.githubusercontent.com/6207498/158638984-3a462302-2a76-402e-bb9a-c77950a8f87a.png)

**Reply by barum** [reply]
Attached is the spreadsheet with merged values and expected results.

[merged-cells.xlsx](https://github.com/jsvine/pdfplumber/files/8265631/merged-cells.xlsx)

**Reply by jsvine** [reply]
Thanks! This definitely helps move the conversation forward. An initial question: Taking this example, why would the final row be duplicated into two rows? I.e., this part:

<img width="228" alt="Screen Shot 2022-03-16 at 4 09 33 PM" src="https://user-images.githubusercontent.com/534702/158699580-8ac51516-7c3b-472e-9916-5d9f143fd0cf.png">

I'm guessing that you prefer the duplication because the row looks like it is double-height. But how might a user, receiving these results, know that those values were *duplicated based on row height* rather than there being two distinct rows?

You don't necessarily have to answer that question, but it's certainly something I'd have to think carefully about in a hypothetical implementation.

**Comment by John-Peter-R** [comment]
Have you got this feature working ?
